### PR TITLE
Allows default credential scopes to be parameterised

### LINKED ISSUE #118: Allow specifying OAuth2 scopes via a property
Right now, we use a static list of scopes for the core credentials. This should be configurable.

### PR DESCRIPTION
**Pull request by joaoandremartins**
Scopes list was static so far. Useful to limit credentials power in
case not every service is used.

Fixes #118

### CODE REVIEW

**Comment by joaoandremartins** [comment]
What do you guys think of introducing the nested `Credentials` class? I'm thinking of following a similar approach with `GcpProperties` by renaming it to `Core` and then nesting it in other starters configurations, then referring to it like

`spring.cloud.gcp.sql.core.project-id`
`spring.cloud.gcp.sql.core.credentials.location`

File: spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/SqlCredentialFactory.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public Credential create() {
 
 		try {
 			return GoogleCredential.fromStream(new FileInputStream(credentialResourceLocation))
-					.createScoped(GcpContextAutoConfiguration.CREDENTIALS_SCOPES_LIST);
+					.createScoped(GcpContextAutoConfiguration.DEFAULT_CREDENTIAL_SCOPES);
```
Comment (meltsufin) [inline spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/SqlCredentialFactory.java]: Why not in this instance use only the scopes needed for Cloud SQL?

Comment (joaoandremartins) [reply spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/SqlCredentialFactory.java]: #121 takes care of that, this is just from the refactor.

**Comment by joaoandremartins** [comment]
Can you think of an alternative name to `core`? That's what GcpProperties will have to be renamed to, if we want to reuse it.

**Comment by meltsufin** [comment]
Unfortunately, I can't think of a better name. :)
That's why I'm not sure about the whole idea of reusing it this way.

**Comment by joaoandremartins** [comment]
@artembilan can you provide your review here? Thanks!

**Pull request merged**